types/signed_header_test.go

```diff
@@ -0,0 +1,104 @@
+package types
+
+import (
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestVerify(t *testing.T) {
+	trusted, privKey, err := GetRandomSignedHeader()
+	require.NoError(t, err)
+	time.Sleep(time.Second)
```

Comment:
😢  why you do me dirty like this and through in bare sleeps in tests haha

What's the need for the sleep? 

If this is a copy paste, let's create a follow up issue to remove so we don't slow down this PR with debugging a potentially flaky test.

Reply:
Sorry, this seemed to be the most straightforward way, but in this case, the sleep is to advance the system time logic so that the next random signed header has a later time (see [code ref](https://github.com/rollkit/rollkit/blob/c7e9725e0fb3bb23f312dad780fa0066333f11d9/types/test_utils.go#L73)), since a verification check in the signed header uses `time.Now()`, modifying the time manually for the new header leads to the error: code [ref.](https://github.com/rollkit/rollkit/blob/c7e9725e0fb3bb23f312dad780fa0066333f11d9/types/header.go#L140C22-L140C59) Anything less than a second leads to running into this error: [code ref](https://github.com/rollkit/rollkit/blob/c7e9725e0fb3bb23f312dad780fa0066333f11d9/types/header.go#L133) since `After` truncates the time to be in seconds.